<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ActionsDialog_q</class>
 <widget class="QWidget" name="ActionsDialog_q">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>852</width>
    <height>229</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Actions Dialog</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <item row="0" column="0">
    <widget class="QFrame" name="frame4">
     <property name="frameShape">
      <enum>QFrame::Shape::Box</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Shadow::Sunken</enum>
     </property>
     <layout class="QGridLayout">
      <item row="0" column="0">
       <widget class="QStackedWidget" name="widgetStack">
        <property name="minimumSize">
         <size>
          <width>320</width>
          <height>0</height>
         </size>
        </property>
        <property name="currentIndex">
         <number>0</number>
        </property>
        <widget class="QWidget" name="NonePage">
         <layout class="QGridLayout">
          <property name="leftMargin">
           <number>12</number>
          </property>
          <property name="topMargin">
           <number>12</number>
          </property>
          <property name="rightMargin">
           <number>12</number>
          </property>
          <property name="bottomMargin">
           <number>12</number>
          </property>
          <property name="spacing">
           <number>12</number>
          </property>
          <item row="0" column="0">
           <widget class="QLabel" name="textLabel5">
            <property name="text">
             <string>This action has no parameters.</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignmentFlag::AlignCenter</set>
            </property>
            <property name="wordWrap">
             <bool>true</bool>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="RejectPage">
         <layout class="QGridLayout">
          <property name="leftMargin">
           <number>12</number>
          </property>
          <property name="topMargin">
           <number>12</number>
          </property>
          <property name="rightMargin">
           <number>12</number>
          </property>
          <property name="bottomMargin">
           <number>12</number>
          </property>
          <property name="spacing">
           <number>12</number>
          </property>
          <item row="1" column="0" colspan="3">
           <widget class="QLabel" name="textLabel3">
            <property name="text">
             <string>If rule action is 'Reject', this option defines firewall's reaction to the packet matching the rule</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignmentFlag::AlignVCenter</set>
            </property>
            <property name="wordWrap">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <spacer>
            <property name="orientation">
             <enum>Qt::Orientation::Vertical</enum>
            </property>
            <property name="sizeType">
             <enum>QSizePolicy::Policy::MinimumExpanding</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>0</height>
             </size>
            </property>
           </spacer>
          </item>
          <item row="2" column="1">
           <widget class="QComboBox" name="rejectvalue">
            <property name="minimumSize">
             <size>
              <width>300</width>
              <height>0</height>
             </size>
            </property>
            <property name="toolTip">
             <string>Defines the firewall's response when rejecting a packet. Maps to iptables --reject-with option.</string>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <spacer>
            <property name="orientation">
             <enum>Qt::Orientation::Horizontal</enum>
            </property>
            <property name="sizeType">
             <enum>QSizePolicy::Policy::MinimumExpanding</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>10</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item row="2" column="2">
           <spacer>
            <property name="orientation">
             <enum>Qt::Orientation::Horizontal</enum>
            </property>
            <property name="sizeType">
             <enum>QSizePolicy::Policy::MinimumExpanding</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>0</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="AccountingStrPage">
         <layout class="QGridLayout">
          <property name="leftMargin">
           <number>12</number>
          </property>
          <property name="topMargin">
           <number>12</number>
          </property>
          <property name="rightMargin">
           <number>12</number>
          </property>
          <property name="bottomMargin">
           <number>12</number>
          </property>
          <property name="spacing">
           <number>12</number>
          </property>
          <item row="1" column="0">
           <widget class="QLabel" name="textLabel1_4">
            <property name="text">
             <string>Rule name for accounting. (white spaces and special characters are not allowed)</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignmentFlag::AlignVCenter</set>
            </property>
            <property name="wordWrap">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLineEdit" name="accountingvalue_str">
            <property name="toolTip">
             <string>Rule name for accounting chain. White spaces and special characters are not allowed.</string>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <spacer>
            <property name="orientation">
             <enum>Qt::Orientation::Vertical</enum>
            </property>
            <property name="sizeType">
             <enum>QSizePolicy::Policy::MinimumExpanding</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>0</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="CustomStrPage">
         <layout class="QVBoxLayout">
          <property name="spacing">
           <number>12</number>
          </property>
          <property name="leftMargin">
           <number>12</number>
          </property>
          <property name="topMargin">
           <number>12</number>
          </property>
          <property name="rightMargin">
           <number>12</number>
          </property>
          <property name="bottomMargin">
           <number>12</number>
          </property>
          <item>
           <widget class="QLabel" name="textLabel1_2">
            <property name="text">
             <string>Custom string:</string>
            </property>
            <property name="wordWrap">
             <bool>false</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="custom_str">
            <property name="toolTip">
             <string>Custom iptables command string appended to the generated rule.</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer>
            <property name="orientation">
             <enum>Qt::Orientation::Vertical</enum>
            </property>
            <property name="sizeType">
             <enum>QSizePolicy::Policy::MinimumExpanding</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>0</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="BranchChainPage">
         <layout class="QGridLayout">
          <property name="spacing">
           <number>12</number>
          </property>
          <item row="0" column="0" colspan="2">
           <widget class="QLabel" name="textLabel1_3">
            <property name="text">
             <string>Policy ruleset object:</string>
            </property>
            <property name="wordWrap">
             <bool>false</bool>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="FWObjectDropArea" name="iptBranchDropArea" native="true">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>100</width>
              <height>80</height>
             </size>
            </property>
            <property name="font">
             <font>
              <pointsize>9</pointsize>
             </font>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QCheckBox" name="ipt_branch_in_mangle">
            <property name="text">
             <string>In addition to 'filter', create branching rule in 'mangle' table as well</string>
            </property>
            <property name="toolTip">
             <string>In addition to 'filter', create branching rule in 'mangle' table as well.</string>
            </property>
           </widget>
          </item>
          <item row="1" column="2">
           <spacer>
            <property name="orientation">
             <enum>Qt::Orientation::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>41</width>
              <height>21</height>
             </size>
            </property>
           </spacer>
          </item>
          <item row="2" column="1">
           <spacer>
            <property name="orientation">
             <enum>Qt::Orientation::Vertical</enum>
            </property>
            <property name="sizeType">
             <enum>QSizePolicy::Policy::MinimumExpanding</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>30</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </widget>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>FWObjectDropArea</class>
   <extends>QWidget</extends>
   <header>firewallfabrik.gui.drop_area</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="MainRes.qrc"/>
 </resources>
</ui>
